docs/git.md

```diff
@@ -7,32 +7,69 @@ One of the most unique features of Simperby is its use of Git.
-Let's assume that there is the last finalized block with the height of `H`.
-We will take that as a starting point of our recursive process. Of course, the base case of the finalized block would be the genesis block.
+## Why Git?
 
-### Step 1: Agenda
+Git is a distributed version control system that is widely used in the software development community. It is a mature technology that has been used for decades.
+There are four main reasons why we use Git as the underlying storage of Simperby.
 
-TODO
+### Version Control System
+
+Git is a version control system, first of all.
+It provides wonderful feature like branching, merging, rebasing, and diffing.
+You can checkout to any commit and browse the file system at that point in time.
+It manages the entire history of the repository in a linear or DAG structure.
+
+This fits amazingly well with Simperby, because.. TODO
+
+### File System
+
+Git manages a file system that can be directly browsed by the users via the host operating system. As Simperby is a blockchain for an organization, not a contract platform, the role of the blockchain state is to serve as a general-purpose data storage. Here are a few examples of the use of a distributed file system for an organization:
+
+1. Replacement of a shared file system like [Google Drive](https://www.google.com/drive/)
+2. Storage for a static fils server for the website that the organization owns.
+3. Codebase of the organization - also the most common use case of Git itself
+4. Diff-sensitive data like the law or the constitution of the organization (articles of association, bylaws, etc.)
+
+If you want to explore the snapshot of the blockchain state (including the past ones), you can simply checkout to the revision and browse the file system.
+
+### Distributed
+
+Git inherently works in a distributed manner.
+A Git repository may have multiple 'remotes' that can be freely added, removed and fetched from. There is no 'central server' for the repository.
+Based on this principle, in Simperby, each node will add the peers that they discovered, and may fetch relevant remote branches, verify them and update their local repository. It could advance the block height by verifying the incoming block headers moving the local `finalized` branch. It could also track newly observed agenda or block proposals provided as peer's remote branches, reflecting on the local repository marked by its own branch.
+
+### Powerful Third-Party Tools and Services
+
+Git is the most used version control system currently.
+It has a huge ecosystem of third-party tools and services.
+
+1. Hosting services like [GitHub](https://github.com) or [GitLab](https://gitlab.com) can easily **mirror** the repository of the organization. This is useful for the organization that wants to provide their data to the public. It will save the cost of developing its own block explorer and indexing service! It is also possible to add a CI plugin to verify the incoming commits as Simperby node does.
```

Comment:
```suggestion
1. Hosting services like [GitHub](https://github.com) or [GitLab](https://gitlab.com) can easily **mirror** the repository of the organization. This is useful for an organization that wants to provide their data to the public. It will save the cost of developing its own block explorer and indexing service! It is also possible to add a CI plugin to verify the incoming commits as Simperby node does.
```